### DISCUSSION TITLE
Templatize and translate email subject as well

### DISCUSSION THREAD

**Discussion by neliets** [InvoiceShelf/InvoiceShelf#169] [Ideas]
**Describe the bug**
Fixed default email subject.

**Expected behavior**
1) Default email subject should be part of translation, perhaps same with text
2) Subject should be possible to templatize (adding variables)

**Screenshots**

**Please complete the following information:**
- InvoiceShelf version: 1.3
- PHP version: not relevant
- Database type and version: not relevant

**Optional info**
- OS:  [e.g. Ubuntu]
- Browser: [e.g. chrome, safari]

For now I will edit the files in my container, but this should be an easy fix, but since I dont know the codebase that well, I currently dont have pull request.

**Comment by mchev** [comment]
I think the issue is resolved, but will only be available in v2.
https://github.com/InvoiceShelf/InvoiceShelf/pull/110

**Comment by rihards-simanovics** [comment]
Broadly speaking I'm not exactly in favour that we store email templates in a PHP file, for best extendibility we should store the template in the DB and populate it using the template strings. This will allow for better extendibility and will reduce any issues that could arise from file modification in source code.

**Comment by neliets** [comment]
> Broadly speaking I'm not exactly in favour that we store email templates in a PHP file, for best extendibility we should store the template in the DB and populate it using the template strings. This will allow for better extendibility and will reduce any issues that could arise from file modification in source code.

I never implied that this would be fixed in php code, just wanted to do a brief work-around, but I was not sucessful.

The issue linked is only talking about translation, but not modifying it and adding variables.

**Comment by mchev** [comment]
The v2 is now available and fix the translation issue.

Yes it could be nice if we can edit the subject. What variables do you need to see in it ? Company name, customer name ?

Maybe this should be transferred to the discussions

**Comment by rihards-simanovics** [comment]
> Maybe this should be transferred to the discussions

Hey @mchev Yeah, I think that's it's a good idea. By the way, sorry didn't clarify in my initial comment, but yes, we can certainly make subject lines more customisable, though I think we should re-think how the email templates are generated in the first place. 

Let me move this to discussion so we can discuss this in more detail as I have some ideas on the current templating system for both Invoices/Estimates and Emails.

**Comment by rihards-simanovics** [comment]
With regards to the templates, I am not a big fan of user having to go into the application source just to make a simple adjustment to an email/invoice/estimate template, it's really not friendly for both UX, and DX as for the user they'd need to know how to code and for Developer it's just extra headaches.

As an example, in Zoho Books they are using WYSIWYG for pretty much everything, and I'd assume store all the templates in the DB for ease of retrieval.
